### PR TITLE
chore: update node version

### PR DESCRIPTION
**Pull request by bryan-robitaille**
# Summary | Résumé

chores:
- update to node 18
- add memory limiter for jest ( [jest docs](https://jestjs.io/docs/configuration#workeridlememorylimit-numberstring) )

# Test instructions | Instructions pour tester la modification

All tests should pass

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [ ] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [ ] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -158,6 +158,7 @@
     "jest-fetch-mock": "^3.0.3",
     "jest-mock-extended": "^3.0.1",
     "next-router-mock": "^0.9.1-beta.0",
+    "node-gyp": "^10.0.1",
```
Comment (timarney) [inline package.json:161]: Is this used somewhere?

Comment (bryan-robitaille) [reply package.json:161]: ah, no... was used for trouble shooting earlier.  Forgot to remove.

**Comment by timarney** [comment]
As part of this update maybe add an .nvmrc file so we can all use the same version within the repo?

https://github.com/nvm-sh/nvm#nvmrc

**Comment by bryan-robitaille** [comment]
> As part of this update maybe add an .nvmrc file so we can all use the same version within the repo?
> 
> https://github.com/nvm-sh/nvm#nvmrc

Great idea!  Will do.

**Comment by craigzour** [comment]
@bryan-robitaille looks like the updated `yarn.lock` included in the pull request is not in sync with the one that is generated locally when I am testing your branch.

**Comment by bryan-robitaille** [comment]
> @bryan-robitaille looks like the updated `yarn.lock` included in the pull request is not in sync with the one that is generated locally when I am testing your branch. I can commit mine if you would like!

yes please.  I didn't update the yarn.lock after removing node-gyp

**Review by craigzour — APPROVED** [review]
LGTM :)

**Comment by timarney** [comment]
Looks good - just re-running the jest tests .

**Review by timarney — APPROVED** [review]

**Pull request merged**